pkg/reconciler/v1alpha1/route/resources/names/names.go

```diff
@@ -31,8 +31,36 @@ func K8sServiceFullname(route *v1alpha1.Route) string {
 	return reconciler.GetK8sServiceFullname(K8sService(route), route.Namespace)
 }
 
-// ClusterIngressPrefix returns GenerateName prefix of the
-// ClusterIngress child resource for given Route.
-func ClusterIngressPrefix(route *v1alpha1.Route) string {
-	return fmt.Sprintf("%s-", route.Name)
+func prefix(s string, l int) string {
+	if l > len(s) {
+		return s
+	}
+	return string(s[0:l])
+}
+
+// ClusterIngressName returns a name that is based on the Route's
+// name, namespace and UID.  The UID is the most important part, because
+// it ensures the uniqueness of the name.  The Route name and namespace
+// is added for human readability.
+func ClusterIngressName(route *v1alpha1.Route) string {
+	// Kubernetes object names cannot exceed 253 characters.
+	totalLen := 253
+	n, ns, uuid := route.Name, route.Namespace, string(route.ObjectMeta.UID)
+	sep := "-"
+	l := totalLen - len(uuid) - len(n) - 2
+	if l > 0 {
+		// Three-parts name: name-prefix(ns)-uuid.
+		name := strings.Join([]string{n, prefix(ns, l), uuid}, sep)
```

Comment:
nit: no need to create a variable `name` here. We can directly return `strings.Join([]string{n, prefix(ns, l), uuid}, sep)`